src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php

```diff
@@ -63,6 +64,94 @@ public function iSortShippingMethodsByName(string $sortType = 'ascending'): void
         ]);
     }
 
+    /**
+     * @When I add the :rule rule configured with :weight
+     */
+    public function iAddTheRuleConfiguredWithWeight(string $rule, int $weight): void
+    {
+        match ($rule) {
+            'Total weight less than or equal' => $this->client->addRequestData('rules', [
+                [
+                    'type' => 'total_weight_less_than_or_equal',
+                    'configuration' => [
+                        'weight' => $weight,
+                    ],
+                ],
+            ]),
+            'Total weight greater than or equal' => $this->client->addRequestData('rules', [
+                [
+                    'type' => 'total_weight_greater_than_or_equal',
+                    'configuration' => [
+                        'weight' => $weight,
+                    ],
+                ],
+            ]),
+            default => throw new \InvalidArgumentException('Unsupported shipping method rule'),
+        };
```

Comment:
```suggestion
        $type = StringInflector::nameToLowercaseCode($rule);
        $this->client->addRequestData('rules', [[
            'type' => $type,
            'configuration' => [
                'weight' => $weight,
            ],
        ]]);
```
and if you prefer to leave the match, refactor it to resolve the `type` by itself and add that into the request data. Similar with `::iAddTheRuleConfiguredWithForChannel`.<br>
Also, after refactoring you could extract the part which adds the data into a private method; the `$this->client->addRequestData('rules', [` is used a lot in here